lib/warehouse/genservers/insert_monitor.ex

```diff
@@ -0,0 +1,160 @@
+defmodule Warehouse.GenServers.InsertMonitor do
```

Comment:
This guy does the lifting of monitoring newly created kits and spawning a cache process for them. This still does poll but is efficient and minimal. Updates are received from HAL but can't do inserts that way since the inserts happen in Joshua. Once we have Shopify, this will be easier to handle.